geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/factory/FastFilterFactory.scala

```diff
@@ -8,28 +8,91 @@
 
 package org.locationtech.geomesa.filter.factory
 
+import org.geotools.feature.simple.SimpleFeatureBuilder
 import org.geotools.filter.text.ecql.ECQL
-import org.locationtech.geomesa.filter.expression.FastPropertyName
+import org.locationtech.geomesa.filter.expression._
+import org.locationtech.geomesa.utils.geotools.{SimpleFeaturePropertyAccessor, SimpleFeatureTypes}
 import org.opengis.feature.`type`.Name
-import org.opengis.filter.expression.PropertyName
-import org.opengis.filter.{Filter, FilterFactory2}
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.MultiValuedFilter.MatchAction
+import org.opengis.filter.expression.{Expression, PropertyName}
+import org.opengis.filter.{Filter, FilterFactory2, PropertyIsEqualTo}
 import org.xml.sax.helpers.NamespaceSupport
 
 /**
- * Filter factory that overrides property name
+  * Filter factory that creates optimized filters
+  *
+  * Note: usage expects the sft to be set in FastFilterFactory.sfts
+  * FastFilterFactory.toFilter will handle this for you
  */
-class FastFilterFactory extends org.geotools.filter.FilterFactoryImpl with FilterFactory2 {
+class FastFilterFactory private extends org.geotools.filter.FilterFactoryImpl with FilterFactory2 {
```

Comment:
:) that is equivalent to having a private constructor - prevents it from being instantiated. Since it relies on the thread local value being set, i wanted to prevent people using it directly without realizinng